server/src/main/java/org/activityinfo/ui/client/page/config/design/ActivityForm.java

```diff
@@ -77,9 +80,17 @@ public ActivityForm(Dispatcher service, UserDatabaseDTO database) {
         add(categoryField);
 
         final MappingComboBox<Integer> locationTypeCombo = new MappingComboBox<Integer>();
-        for (LocationTypeDTO type : database.getCountry().getLocationTypes()) {
-            locationTypeCombo.add(type.getId(), type.getName());
-        }
+        database.then(new Function<UserDatabaseDTO, Object>() {
```

Comment:
What happens if the request fails?
Whenever a combo box is populated by a remote source, we need to address the case where the request fails. If the request fails here, then the form will be in invalid state and the user will be confused.

This is the same problem we have with the GWTChosen widget, but here at least with GXT we have the right component: it should be RemoteComboBox with an asynchronous loader.